### PR TITLE
HTML-681: enrollInProgram tag should set the location

### PR DESCRIPTION
**Pull request by mogoodrich**
## Issue I worked on

see https://issues.openmrs.org/browse/HTML-

## Checklist: I completed these to help reviewers :)

- [ ] My pull request only contains **ONE single commit**
(the number above, next to the 'Commits' tab is 1).

  No? -> [read here](https://wiki.openmrs.org/display/docs/Pull+Request+Tips) on how to squash multiple commits into one

- [ ] My IDE is configured to follow the [**code style**](https://wiki.openmrs.org/display/docs/Java+Conventions) of this project.

  No? Unsure? -> [configure your IDE](https://wiki.openmrs.org/display/docs/How-To+Setup+And+Use+Your+IDE), format the code and add the changes with `git add . && git commit --amend`

- [ ] I have **added tests** to cover my changes. (If you refactored
  existing code that was well tested you do not have to add tests)

  No? -> write tests and add them to this commit `git add . && git commit --amend`

- [ ] I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.

  No? -> execute above command

- [ ] All new and existing **tests passed**.

  No? -> figure out why and add the fix to your commit. It is your responsibility to make sure your code works.

- [ ] My pull request is **based on the latest changes** of the master branch.

  No? Unsure? -> execute command `git pull --rebase upstream master`

### CODE REVIEW

**Comment by mogoodrich** [comment]
Sorry, a fair amount of formatting changes appeared in this PR, but shouldn't be too hard to find the actual changes.

**Comment by mogoodrich** [comment]
And @mks-d  more an fyi for you... not sure if you use the enrollInProgram functionality at all

File: api/src/main/java/org/openmrs/module/htmlformentry/FormSubmissionActions.java [hunk 1]
```diff
@@ -374,26 +372,31 @@ public void modifyObs(Obs existingObs, Concept concept, Object newValue, Date ne
...
 	 * enrollment date as the date specified and setting initial states from the specified state
 	 * <p/>
 	 * Note that this method does not commit the program enrollment to the database but instead adds the
 	 * Program to a list of programs to add. The changes are applied elsewhere in the framework
-	 * 
+	 *
 	 * @param program Program to enroll the patient in
 	 * @param enrollmentDate the date to enroll the patient in the program
 	 * @param states list of states to set as initial in their workflows
 	 */
```
Comment (mseaton) [inline api/src/main/java/org/openmrs/module/htmlformentry/FormSubmissionActions.java:397]: Javadoc on locationTag parameter?  What is this for?

File: api/src/main/java/org/openmrs/module/htmlformentry/HtmlFormEntryUtil.java [hunk 2]
```diff
@@ -1165,6 +1164,24 @@ public static LocationTag getLocationTag(String identifier) {
...
+	/**
+	 * Finds the first ancestor (including the current location) that is tagged with the specified
+	 * location tag
+	 *
+	 * @param location
+	 * @param locationTag
+	 * @return
+	 */
+	public static Location getFirstAncestorWithTag(Location location, LocationTag locationTag) {
+		if (location.hasTag(locationTag.getName())) {
```
Comment (mseaton) [inline api/src/main/java/org/openmrs/module/htmlformentry/HtmlFormEntryUtil.java:1176]: Might want a not-null check on location here, just in case.

**Review by mseaton — APPROVED** [review]
Looks good, a few minor things to ideally address.

**Comment by mogoodrich** [comment]
Thanks @mseaton ... will address.

Looks like I forgot to push up the new test form I added as well (which resulted in failing Travis tests).  Just corrected that.

**Comment by mogoodrich** [comment]
Addressed PR comments, merging, thanks @mseaton !

**Pull request merged**